tests/Zend/Validate/DateTest.php

```diff
@@ -208,7 +208,11 @@ public function testLocaleContructor()
      */
     public function testNonStringValidation()
     {
-        $this->assertFalse($this->_validator->isValid(array(1 => 1)));
+        try {
+            $this->assertFalse($this->_validator->isValid(array(1 => 1)));
+        } catch (Error $e) {
+            $this->assertTrue($e instanceof TypeError);
+        }
```

Comment:
if validators throw `TypeError`s in PHP8 instead of returning boolean value, I suppose it's a different behavior now? Could it be adjusted in the validator's base class so that it does not throw? Or was the error behavior also expected in php < 8?

Reply:
The TypeError isn't thrown by any validator. It's due to using an array as the first argument in `iconv_strpos`. Before 8 this just emitted a warning and returned a false.

Reply:
but `iconv_strpos` is called inside the date validator, isn't it? Perhaps we could check the type of the value to be validated inside the validator and return false if type is wrong, to avoid the `TypeError` to be thrown and have the behavior consistent among all php versions? 🤔

Reply:
OK. In `Zend_Validate_Date::_checkFormat` exceptions are already caught and the method returns false. This just needed to be extended to also catch errors. Test has been reverted.